gapic/src/main/com/google/gapid/perfetto/views/PowerSummaryPanel.java

```diff
@@ -0,0 +1,279 @@
+import static com.google.gapid.perfetto.views.StyleConstants.mainGradient;
+
+import com.google.gapid.perfetto.Unit;
+import com.google.gapid.perfetto.canvas.Area;
+import com.google.gapid.perfetto.canvas.Fonts;
+import com.google.gapid.perfetto.canvas.RenderContext;
+import com.google.gapid.perfetto.canvas.Size;
+import com.google.gapid.perfetto.models.PowerSummaryTrack;
+import com.google.gapid.perfetto.models.Selection;
+
+public class PowerSummaryPanel extends TrackPanel<PowerSummaryPanel> {
+  private static final double HOVER_MARGIN = 10;
+  private static final double HOVER_PADDING = 4;
+  private static final double CURSOR_SIZE = 5;
+  private static Unit unit;
+
+  protected final PowerSummaryTrack track;
+  protected static final double trackHeight = 80;
+  protected HoverCard hovered = null;
+
+  public PowerSummaryPanel(State state, PowerSummaryTrack track) {
+    super(state);
+    this.track = track;
+    this.unit = track.unit;
+  }
+
+  @Override
+  public PowerSummaryPanel copy() {
+    return new PowerSummaryPanel(state, track);
+  }
+
+  @Override
+  public String getTitle() {
+    return "Power Usage";
+  }
+
+  @Override
+  public String getSubTitle() {
+    return track.getNumPowerRailTracks() + " power rail tracks";
+  }
+
+  @Override
+  public String getTooltip() {
+    return "Total Power Usage";
+  }
+
+  @Override
+  public double getHeight() {
+    return trackHeight;
+  }
+
+  @Override
+  protected void renderTrack(RenderContext ctx, Repainter repainter, double w, double h) {
+    ctx.trace(
+        "PowerSummary",
+        () -> {
+          PowerSummaryTrack.Data data = track.getData(state.toRequest(), onUiThread(repainter));
+          drawLoading(ctx, data, state, h);
+
+          if (data == null || data.ts.length == 0) {
+            return;
+          }
+
+          double range =
+              (track.minValue == track.maxValue && track.minValue == 0)
+                  ? 1
+                  : (track.maxValue - track.minValue);
+
+          Selection<?> selected = state.getSelection(Selection.Kind.Counter);
+          mainGradient().applyBaseAndBorder(ctx);
+          ctx.path(
+              path -> {
+                double lastX = state.timeToPx(data.ts[0]), lastY = h;
+                path.moveTo(lastX, lastY);
+                for (int i = 0; i < data.ts.length; i++) {
+                  double nextX = state.timeToPx(data.ts[i]);
+                  double nextY =
+                      (trackHeight - 1) * (1 - (data.values[i] - track.minValue) / range);
+                  path.lineTo(nextX, lastY);
+                  path.lineTo(nextX, nextY);
+                  lastX = nextX;
+                  lastY = nextY;
+                }
+
+                path.lineTo(lastX, h);
+                ctx.fillPath(path);
+                ctx.drawPath(path);
+              });
+
+          if (hovered != null) {
+            double y = (trackHeight - 1) * (1 - (hovered.value - track.minValue) / range);
```

Comment:
nit: The formula used to derive `y` here and in line 98 (`(trackHeight - 1) * (1 - (x - track.minValue) / range)`) looks like it could be extracted to its own private function just to make it easier to parse (since it's also used in line 98)

Reply:
Could be. Again, that's how the calculation has been done in other Views, but personally I agree that it would be neater to have a function. I can made that change.